### PR TITLE
Entra ID support and azd bicep deployment files for Aspnet core quickstart

### PR DESCRIPTION
**Pull request by CawaMS**
## Purpose

* ...

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: quickstart/aspnet-core/README.md [hunk 1]
```diff
@@ -11,47 +11,43 @@ products:
 ---
 # Quickstart: Use Azure Cache for Redis with an ASP.NET Core web app
 
-This sample shows you how to use an ASP.NET Core web application to connect to Azure Cache for Redis to store and retrieve data from the cache. See the [accompanying article](https://docs.microsoft.com/azure/azure-cache-for-redis/cache-web-app-aspnet-core-howto) on the documentation site for details, including best practices and how to create the sample code from scratch.
+This sample shows you how to use an ASP.NET Core web application to connect to Azure Cache for Redis to store and retrieve data from the cache. Microsoft Entra ID is used for connecting the web application to the Azure Cache for Redis. See the [accompanying article](https://docs.microsoft.com/azure/azure-cache-for-redis/cache-web-app-aspnet-core-howto) on the documentation site for details, including best practices and how to create the sample code from scratch.
```
Comment (philon-msft) [inline quickstart/aspnet-core/README.md:14]: "Microsoft Entra ID is used to authenticate the connection to the Redis cache"

File: quickstart/aspnet-core/infra/core/database/sqlserver/sqlserver.bicep [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    administratorLogin: sqlAdmin
+    administratorLoginPassword: sqlAdminPassword
+  }
+
+  resource database 'databases' = {
+    name: databaseName
+    location: location
+  }
+
+  // resource firewall 'firewallRules' = {
```
Comment (philon-msft) [inline quickstart/aspnet-core/infra/core/database/sqlserver/sqlserver.bicep:36]: remove commented-out block?

File: quickstart/aspnet-core/infra/core/ai/cognitiveservices.bicep [hunk 3]
```diff
@@ -0,0 +1,42 @@
+metadata description = 'Creates an Azure Cognitive Services instance.'
```
Comment (philon-msft) [inline quickstart/aspnet-core/infra/core/ai/cognitiveservices.bicep:1]: This PR is adding a huge amount of bicep. Who's going to maintain it? Is this the minimum amount needed to demonstrate Redis concepts?

**Review by philon-msft — CHANGES_REQUESTED** [review]

**Comment by CawaMS** [comment]
@philon-msft my latest commit resolves all your comments. I also tested to ensure everything works. Can you please review and approve?

File: quickstart/aspnet-core/infra/core/database/sqlserver/sqlserver.bicep (hunk 2 above)
Comment (CawaMS) [reply quickstart/aspnet-core/infra/core/database/sqlserver/sqlserver.bicep]: removed

File: quickstart/aspnet-core/ContosoTeamStats/RedisConnection.cs [hunk 4]
```diff
@@ -143,7 +159,12 @@ private async Task ForceReconnectAsync(bool initializing = false)
                 _previousErrorTime = DateTimeOffset.MinValue;
 
                 // Create a new connection
-                ConnectionMultiplexer _newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
+                //ConnectionMultiplexer _newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
+
+                var configurationOptions = await ConfigurationOptions.Parse($"{_redisHostName}:6380").ConfigureForAzureWithSystemAssignedManagedIdentityAsync();
```
[the remark this answers is not in the document]
Comment (CawaMS) [reply quickstart/aspnet-core/ContosoTeamStats/RedisConnection.cs]: fixed.

File: quickstart/aspnet-core/ContosoTeamStats/RedisConnection.cs [hunk 5]
```diff
@@ -25,18 +27,32 @@ public class RedisConnection : IDisposable
...
 
         private SemaphoreSlim _reconnectSemaphore = new SemaphoreSlim(initialCount: 1, maxCount: 1);
-        private readonly string _connectionString;
+        //private readonly string _connectionString;
+        private readonly string _redisHostName;
         private ConnectionMultiplexer _connection;
         private IDatabase _database;
 
-        private RedisConnection(string connectionString)
+        //private RedisConnection(string connectionString)
```
[the remark this answers is not in the document]
Comment (CawaMS) [reply quickstart/aspnet-core/ContosoTeamStats/RedisConnection.cs]: cleaned up

File: quickstart/aspnet-core/ContosoTeamStats/ContosoTeamStats.csproj [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
     <UserSecretsId>ContosoTeamStats</UserSecretsId>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Web.LibraryManager.Build" Version="2.1.113" />
+    <PackageReference Include="Microsoft.Azure.StackExchangeRedis" Version="3.1.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Web.LibraryManager.Build" Version="2.1.175" />
     <PackageReference Include="StackExchange.Redis" Version="2.*" />
```
[the remark this answers is not in the document]
Comment (CawaMS) [reply quickstart/aspnet-core/ContosoTeamStats/ContosoTeamStats.csproj]: fixed.

**Comment by CawaMS** [comment]
@philon-msft would you mind approving this PR? I have not seen additional comment from the reviewers since my latest commits.

File: quickstart/aspnet-core/azure.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
```
Comment (stanleysmall-microsoft) [inline quickstart/aspnet-core/azure.yaml:1]: Why do we have so many commented lines in this file? Do we need them?

File: quickstart/aspnet-core/infra/abbreviations.json [hunk 8]
```diff
@@ -0,0 +1,136 @@
+{
```
Comment (stanleysmall-microsoft) [inline quickstart/aspnet-core/infra/abbreviations.json:1]: Do we need this file? It seems to add complexity when we only use it a couple of times

File: quickstart/aspnet-core/infra/main.bicep [hunk 9]
```diff
@@ -0,0 +1,70 @@
+targetScope = 'subscription'
+
+// The main bicep module to provision Azure resources.
+// For a more complete walkthrough to understand how this file works with azd,
+// see https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/make-azd-compatible?pivots=azd-create
```
Comment (stanleysmall-microsoft) [inline quickstart/aspnet-core/infra/main.bicep:5]: Remove `en-us/`

File: quickstart/aspnet-core/infra/resources.bicep [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+    httpsOnly: true
+  }
+  identity: {
+    type: 'SystemAssigned'
+  }
+  
+  resource appSettings 'config' = {
+    name: 'appsettings'
+    properties: {
+      //"ENABLE_ORYX_BUILD" : "false", "SCM_DO_BUILD_DURING_DEPLOYMENT" : "false",
```
Comment (stanleysmall-microsoft) [inline quickstart/aspnet-core/infra/resources.bicep:139]: Remove this comment?

File: quickstart/aspnet-core/ContosoTeamStats/ContosoTeamStats.csproj [hunk 11]
```diff
@@ -1,14 +1,15 @@
...
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <UserSecretsId>ContosoTeamStats</UserSecretsId>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Web.LibraryManager.Build" Version="2.1.113" />
-    <PackageReference Include="StackExchange.Redis" Version="2.*" />
+    <PackageReference Include="Microsoft.Azure.StackExchangeRedis" Version="3.1.0" />
```
Comment (philon-msft) [inline quickstart/aspnet-core/ContosoTeamStats/ContosoTeamStats.csproj:9]: Let's reference version `3.*` to automatically pick up latest

File: quickstart/aspnet-core/azure.yaml (hunk 7 above)
Comment (CawaMS) [reply quickstart/aspnet-core/azure.yaml:1]: fixed

File: quickstart/aspnet-core/infra/abbreviations.json (hunk 8 above)
Comment (CawaMS) [reply quickstart/aspnet-core/infra/abbreviations.json]: removed this file

File: quickstart/aspnet-core/infra/main.bicep (hunk 9 above)
Comment (CawaMS) [reply quickstart/aspnet-core/infra/main.bicep]: removed the comments in main.bicep to keep the content minimal.

File: quickstart/aspnet-core/infra/resources.bicep (hunk 10 above)
Comment (CawaMS) [reply quickstart/aspnet-core/infra/resources.bicep]: removed

File: quickstart/aspnet-core/ContosoTeamStats/ContosoTeamStats.csproj (hunk 11 above)
Comment (CawaMS) [reply quickstart/aspnet-core/ContosoTeamStats/ContosoTeamStats.csproj:9]: fixed

**Review by stanleysmall-microsoft — APPROVED** [review]

**Pull request merged**